main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -2261,11 +2265,17 @@ object Typer {
         val ms = methods map visitJvmMethod
         TypedAst.Expression.NewObject(name, clazz, tpe, pur, eff, ms, loc)
 
-      case KindedAst.Expression.NewChannel(exp, elmTpe, loc) =>
+      case KindedAst.Expression.NewChannel(reg, exp, elmTpe, loc) =>
+        val r = visitExp(reg, subst0)
         val e = visitExp(exp, subst0)
         val pur = Type.Impure
         val eff = e.eff
-        TypedAst.Expression.NewChannel(e, Type.mkTuple(List(Type.mkSender(elmTpe, Type.False, loc), Type.mkReceiver(elmTpe, Type.False, loc)), loc), elmTpe, pur, eff, loc)
+        r.tpe match {
+          case Type.Apply(_, regVar, _) => 
+            TypedAst.Expression.NewChannel(r, e, Type.mkTuple(List(Type.mkSender(elmTpe, regVar, loc), Type.mkReceiver(elmTpe, regVar, loc)), loc), elmTpe, pur, eff, loc)
+          case _ => 
+            throw new InternalCompilerException("Unexpected region type")
```

Comment:
@magnus-madsen I'd love to avoid this if possible. Is there any way to do so?

Reply:
Yeah, this is no good. I think you really just want a type variable in NewChan that you unify with the result and reuse here.

Reply:
Ah right, got you. Which I think answers our question in the previous PR. Will do.